modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/infrastructure/consensus/ConsensusManager.scala

```diff
@@ -198,47 +211,78 @@ object ConsensusManager {
         trigger: Option[ConsensusTrigger]
       ): F[Unit] =
         consensusStorage.getLastConsensusOutcome.flatMap { maybeLastOutcome =>
-          maybeLastOutcome.traverse { lastOutcome =>
-            val nextKey = _key.get(lastOutcome).next
-
-            consensusStorage
-              .getResources(nextKey)
-              .flatMap { resources =>
-                logger.debug(s"Trying to facilitate consensus {key=${nextKey.show}, trigger=${trigger.show}}") >>
-                  consensusStateCreator.tryFacilitateConsensus(nextKey, lastOutcome, trigger, resources).flatMap {
-                    case Some(state) =>
-                      stallDetection(nextKey, state) >>
-                        internalCheckForStateUpdate(nextKey, resources)
-                    case None => Applicative[F].unit
-                  }
-              }
-          }.void
+          maybeLastOutcome match {
+            case None =>
+              logger.trace(s"No last consensus outcome available, cannot facilitate")
+            case Some(lastOutcome) =>
+              val nextKey = _key.get(lastOutcome).next
+
+              consensusStorage
+                .getResources(nextKey)
+                .flatMap { resources =>
+                  logger.debug(s"Trying to facilitate consensus {key=${nextKey.show}, trigger=${trigger.show}}") >>
+                    consensusStateCreator.tryFacilitateConsensus(nextKey, lastOutcome, trigger, resources).flatMap {
+                      case Some(state) =>
+                        // logger.debug(s"Created new consensus state for {key=${nextKey.show}}") >>
+                        stallDetection(nextKey, state) >>
+                          internalCheckForStateUpdate(nextKey, resources)
+                      case None =>
+                        logger.trace(s"Could not create consensus state for {key=${nextKey.show}}") >>
+                          // Even if we can't create a new state, check if existing state needs updating
+                          consensusStorage.getState(nextKey).flatMap {
+                            case Some(_) =>
+                              logger.trace(s"Found existing state for {key=${nextKey.show}}, checking for updates") >>
+                                internalCheckForStateUpdate(nextKey, resources)
+                            case None =>
+                              logger.trace(s"No existing state for {key=${nextKey.show}}, nothing to do") >>
+                                Applicative[F].unit
+                          }
+                    }
+                }
+          }
```

Comment:
This is same logic just lots of tracing statements to figure out why the stall trigger code was never being reached.